include/library/spdm_requester_lib.h

```diff
@@ -731,7 +734,7 @@ libspdm_return_t libspdm_get_csr(void * context,
                                  void * requester_info, uint16_t requester_info_length,
                                  void * opaque_data, uint16_t opaque_data_length,
                                  const uint32_t *session_id,
-                                 void * csr, size_t csr_max_len);
+                                 void **csr, size_t *csr_len);
```

Comment:
I think the interface should be `void *csr, size_t *csr_len`

on input, *csr_len indicates the max csr buffer size.
on output, *csr_len indicates the actual csr buffer size.

We still need the caller to allocate the buffer, instead of the callee.